packages/flutter/lib/src/material/list.dart

```diff
@@ -22,12 +23,73 @@ Map<MaterialListType, double> kListItemExtent = const <MaterialListType, double>
   MaterialListType.threeLine: kThreeLineListItemHeight,
 };
 
+class _Painters extends ScrollableListPainter {
```

Comment:
I'd probably move this next to ScrollableListPainter and make it public.  CompoundScrollableListPainter ?